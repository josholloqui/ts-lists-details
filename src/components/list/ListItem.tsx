import * as React from 'react';
import { Link } from 'react-router-dom';


interface Props {
  characters: {char_id: number, name: string}[]
}

const ListItem: React.FunctionComponent<Props> = ({ characters }) => {
  return (
    <ul>
      {characters.map(character => (
        <li key={character.char_id}>
          <Link to={`/details/${character.char_id}`} >
            <h2>{character.name}</h2>
          </Link>
        </li>
      ))}
    </ul>
  )
}

export default ListItem;
