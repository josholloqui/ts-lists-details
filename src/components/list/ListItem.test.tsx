import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import ListItem from './ListItem';
import { MemoryRouter } from 'react-router';

describe('ListItem component', () => {
  afterEach(() => cleanup());

  it('render ListItem', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <ListItem
          characters={[
            {
              char_id: 1,
              name: 'fred'
            },
            {
              char_id: 2,
              name: 'dave'
            }
          ]}
        />
      </MemoryRouter>
      
    );

    expect(asFragment()).toMatchSnapshot();
  })
})
