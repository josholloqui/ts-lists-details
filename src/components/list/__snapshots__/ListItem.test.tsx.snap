// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component render ListItem 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <a
        href="/details/1"
      >
        <h2>
          fred
        </h2>
      </a>
    </li>
    <li>
      <a
        href="/details/2"
      >
        <h2>
          dave
        </h2>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
